File: asignaturas/sql/Asignaturas-IvanRonco/Profesores_Y_Modulos_BD-Java/src/main/java/mapper/modulo/ModuloMapper.java
package mapper.modulo;

import dto.modulo.ListaModulosDto;
import dto.modulo.ModuloDto;
import models.Modulo;
import utils.Utils;

import java.util.List;
import java.util.UUID;

public class ModuloMapper {
    public static ModuloDto toModuloDto(Modulo modulo){
        return new ModuloDto(
                modulo.getUuid().toString(),
                modulo.getNombre(),
                String.valueOf(modulo.getCurso()),
                modulo.getGrado().toString()
        );
    }

    public static Modulo toModulo(ModuloDto moduloDto){
        return new Modulo(
                UUID.fromString(moduloDto.getUuid()),
                moduloDto.getNombre(),
                Integer.parseInt(moduloDto.getCurso()),
                Utils.toEnumGrado(moduloDto.getGrado())
        );
    }

    public static ListaModulosDto toModulosDto(List<Modulo> modulos){
        return new ListaModulosDto(
                modulos.stream().map(it -> toModuloDto(it)).toList()
        );
    }

    public static List<Modulo> toModulos(ListaModulosDto modulosDto){
        return modulosDto.getModuloDtos().stream().map(it -> toModulo(it)).toList();
    }
}

File: asignaturas/sql/SergioDeIscar-Asignatura-Kotlin/SergioDeIscar-Asignatura-Kotlin/src/main/kotlin/mappers/ModuloMapper.kt
package mappers

import dto.ModuloDto
import models.Grado
import models.Modulo
import java.util.*

fun Modulo.toDto(): ModuloDto{
    return ModuloDto(
        uuid.toString(),
        nombre,
        curso.toString(),
        grado.toString()
    )
}

fun ModuloDto.toClass(): Modulo{
    return Modulo(
        UUID.fromString(uuid),
        nombre,
        curso.toInt(),
        Grado.valueOf(grado)
    )
}